lib/beaker/host.rb

```diff
@@ -224,13 +225,29 @@ def is_x86_64?
     end
 
     def connection
-      @connection ||= SshConnection.connect( reachable_name,
+      # create new connection object if necessary
+      @connection ||= SshConnection.connect( { :ip => self['ip'], :vmhostname => self['vmhostname'], :hostname => @name },
```

Comment:
Ah, that's tricky.  If you call self.ip it will attempt to get the ip through a host.exec - which then attempts to get a connection object to the host which then attempts to determine the ip and so on until stack too deep.

So, we only use the ip if it is known, either set by the provisioner or determined by the user after a connection has been established.